scraper/ordoclic.py

```diff
@@ -0,0 +1,127 @@
+    headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+    r = session.post(base_url, data=json.dumps(payload), headers=headers)
+    r.raise_for_status()
+    return(r.json())
+
+def getProfile(rdv_site_web):
+    slug = rdv_site_web.rsplit('/', 1)[-1]
+    prof = rdv_site_web.rsplit('/', 2)[-2]
+    if prof in ['pharmacien', 'medecin']:
+        base_url = f'https://api.ordoclic.fr/v1/professionals/profile/{slug}'
+    else:
+        base_url = f'https://api.ordoclic.fr/v1/public/entities/profile/{slug}'
+    r = session.get(base_url)
+    r.raise_for_status()
+    return(r.json())
+
+def parse_ordoclic_slots(availability_data):
+    first_availability = None
+    if not availability_data:
+        return None
+    if 'nextAvailableSlotDate' in availability_data:
+        nextAvailableSlotDate = availability_data.get('nextAvailableSlotDate', None)
+        if nextAvailableSlotDate != None:
+            first_availability = datetime.strptime(nextAvailableSlotDate, '%Y-%m-%dT%H:%M:%S%z')
+            first_availability += first_availability.replace(tzinfo=timezone('CET')).utcoffset()
+            return first_availability
+
+    availabilities = availability_data.get('slots', None)
+    if availabilities is None:
+        return None
+    for slot in availabilities:
+        timeStart = slot.get('timeStart', None)
+        if not timeStart:
+            continue
+        date = datetime.strptime(timeStart, '%Y-%m-%dT%H:%M:%S%z')
+        if 'timeStartUtcOffset' in slot:
+            timeStartUtcOffset = slot["timeStartUtcOffset"]
+            date += timedelta(minutes=timeStartUtcOffset)
+        if first_availability is None or date < first_availability:
+            first_availability = date        
+    return first_availability
+    
+def fetch_slots(rdv_site_web, start_date):
+    first_availability = None    
+    profile = getProfile(rdv_site_web)
+    slug = profile["profileSlug"]
+    entityId = profile["entityId"]
+    for professional in profile["publicProfessionals"]:
+        medicalStaffId = professional["id"]
+        name = professional["fullName"]
+        zip = professional["zip"]
+        reasons = getReasons(entityId)
+        #reasonTypeId = 4 -> 1er Vaccin
+        for reason in reasons["reasons"]:
+            if reason["reasonTypeId"] == 4 and reason["canBookOnline"] == True: 
+                reasonId = reason["id"]
+                date_obj = datetime.strptime(start_date, '%Y-%m-%d')
+                end_date = (date_obj + timedelta(days=6)).strftime('%Y-%m-%d')
+                slots = getSlots(entityId, medicalStaffId, reasonId, start_date, end_date)
+                date = parse_ordoclic_slots(slots)
+                if date is None:
+                    continue
+                if first_availability is None or date < first_availability:
+                    first_availability = date
+    if first_availability == None:
+        return None
+    return str(first_availability)
```

Comment:
On s'attend à une date au format ISO en retour, il faut donc faire :
```suggestion
    return first_availability.isoformat()
```